ngraph/src/ngraph/frontend/onnx_import/op/instance_norm.cpp

```diff
@@ -41,47 +41,89 @@ namespace ngraph
+                    Output<ngraph::Node> data(node.get_ng_inputs().at(0));
                     Output<ngraph::Node> scale(node.get_ng_inputs().at(1));
                     Output<ngraph::Node> bias(node.get_ng_inputs().at(2));
-                    const Shape& data_shape = data->get_shape();
-                    const Shape& scale_shape = scale.get_shape();
-                    const Shape& bias_shape = bias.get_shape();
+                    const PartialShape& data_pshape = data.get_partial_shape();
+                    const PartialShape& scale_pshape = scale.get_partial_shape();
+                    const PartialShape& bias_pshape = bias.get_partial_shape();
                     const float epsilon{node.get_attribute_value<float>("epsilon", 1e-5f)};
 
+                    element::Type result_et;
                     CHECK_VALID_NODE(
                         node,
-                        (scale_shape.size() == 1 && scale_shape[0] == data_shape.at(1)),
-                        "Scale input must be one dimensional vector of number of "
-                        "input data channels size.");
+                        element::Type::merge(
+                            result_et, data.get_element_type(), scale.get_element_type()),
+                        "Element types for data and scale input do not match (data element type: ",
+                        data.get_element_type(),
+                        ", scale element type: ",
+                        scale.get_element_type(),
+                        ").");
+
+                    CHECK_VALID_NODE(
+                        node,
+                        element::Type::merge(
+                            result_et, data.get_element_type(), bias.get_element_type()),
+                        "Element types for data and bias input do not match (data element type: ",
+                        data.get_element_type(),
+                        ", bias element type: ",
+                        bias.get_element_type(),
+                        ").");
 
                     CHECK_VALID_NODE(node,
-                                     (bias_shape.size() == 1 && bias_shape[0] == data_shape.at(1)),
+                                     (scale_pshape.rank().is_static() &&
+                                      scale_pshape.rank().get_length() == 1 &&
+                                      data_pshape.rank().is_static() &&
+                                      data_pshape[1].same_scheme(scale_pshape[0])),
+                                     "Scale input must be one dimensional vector of number of "
+                                     "input data channels size.");
+
+                    CHECK_VALID_NODE(node,
+                                     (bias_pshape.rank().is_static() &&
+                                      bias_pshape.rank().get_length() == 1 &&
+                                      data_pshape.rank().is_static() &&
```

Comment:
Maybe we should make these checks more relaxed to enable also dynamic data shape rank.